File: Server/playpredict/prediction.py
import json


class Prediction:
    def __init__(self, formation, play, confidence):
        self.formation = formation
        self.play = play
        self.confidence = confidence

    def toJSON(self):
        return json.dumps(self, default=lambda o: o.__dict__, sort_keys=True, indent=4)

File: app/playpredict/lib/models/prediction.dart
// ignore_for_file: non_constant_identifier_names

import 'package:json_annotation/json_annotation.dart';

part 'prediction.g.dart';

@JsonSerializable()
class Prediction {
  Prediction({
    required this.formation,
    required this.image,
    required this.name,
    required this.weight,
  });

  factory Prediction.fromJson(Map<String, dynamic> json) =>
      _$PredictionFromJson(json);

  final String formation;
  final String image;
  final String name;
  final double weight;

  Map<String, dynamic> toJson() => _$PredictionToJson(this);
}
